src/main/java/com/amazon/ion/impl/IonReaderContinuableCore.java

```diff
@@ -35,6 +39,12 @@ interface IonReaderContinuableCore extends IonCursor {
      */
     IonType getType();
 
+    /**
+     * Returns the type of the current value in the raw encoding, or
+     * null if there is no current value.
+     */
+    IonType getEncodingType();
```

Comment:
Higher-level readers override this; this method gives us a way to get at the low-level encoding type no matter what the concrete type of the reader instance actually is.

Reply:
Probably, though that will be addressed in the forthcoming PR that adds macro evaluation.